x/oracle/abci.go

```diff
@@ -29,22 +31,34 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 			continue
 		}
 
-		ctx.EventManager().EmitEvent(
-			sdk.NewEvent(
-				"aggregate_task",
-				sdk.NewAttribute("contract", task.Contract),
-				sdk.NewAttribute("function", task.Function),
-				sdk.NewAttribute("begin_block_height", strconv.FormatInt(task.BeginBlock, 10)),
-				sdk.NewAttribute("bounty", task.Bounty.String()),
-				sdk.NewAttribute("description", task.Description),
-				sdk.NewAttribute("expiration", task.Expiration.String()),
-				sdk.NewAttribute("creator", task.Creator),
-				sdk.NewAttribute("responses", task.Responses.String()),
-				sdk.NewAttribute("result", task.Result.String()),
-				sdk.NewAttribute("end_block_height", strconv.FormatInt(task.ClosingBlock, 10)),
-				sdk.NewAttribute("status", task.Status.String()),
-			),
-		)
+		switch task := task.(type) {
+		case *types.Task:
+			ctx.EventManager().EmitEvent(
+				sdk.NewEvent(
+					"aggregate_task",
+					sdk.NewAttribute("contract", task.Contract),
+					sdk.NewAttribute("function", task.Function),
+					sdk.NewAttribute("begin_block_height", strconv.FormatInt(task.BeginBlock, 10)),
+					sdk.NewAttribute("bounty", task.Bounty.String()),
+					sdk.NewAttribute("description", task.Description),
+					sdk.NewAttribute("expiration", task.Expiration.String()),
+					sdk.NewAttribute("creator", task.Creator),
+					sdk.NewAttribute("responses", task.Responses.String()),
+					sdk.NewAttribute("result", task.Result.String()),
+					sdk.NewAttribute("end_block_height", strconv.FormatInt(task.ClosingBlock, 10)),
+					sdk.NewAttribute("status", task.Status.String()),
+				),
+			)
+		case *types.TxTask:
+			//implement me
+			ctx.EventManager().EmitEvent(
+				sdk.NewEvent(
+					"aggregate_task",
+					//shall be base64 encoded to be aligned with proto json encoding
+					sdk.NewAttribute("tx_hash", base64.StdEncoding.EncodeToString(task.TxHash)),
```

Comment:
hex.EncodeToString instead of base64.StdEncoding.EncodeToString?

Reply:
I guess we'd better do Base64 because that's aligned with protobuf json encoding.